### PR TITLE
Request code review for game logic

### PR DESCRIPTION
**Pull request by kobiyoyo**

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    def switch_players
+      @current_player, @other_player = @other_player, @current_player
+    end
+
+    def solicit_move
+      "#{current_player.name}: Enter a number between 1 and 9 to make your move"
+    end
+
+    def get_move(human_move = gets.chomp)
```
Comment (bellom) [inline lib/game.rb]: Game logic should be kept in classes (no “puts” and “get.chomp” inside the `game.rb`)

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    public
+
+    def game_over_message
+      return "#{current_player.name} won!" if board.game_over == :winner
+      return "The game ended in a tie" if board.game_over == :draw
+    end
+
+    def play
+      puts "#{current_player.name} has randomly been selected as the first player"
```
Comment (bellom) [inline lib/game.rb]: Game logic should be kept in classes (no “puts” and “get.chomp” inside the `game.rb`)

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      while true
+        board.formatted_grid
+        puts ""
+        puts solicit_move
```
Comment (bellom) [inline lib/game.rb]: Game logic should be kept in classes (no “puts” and “get.chomp” inside the `game.rb`)

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        x, y = get_move
+        board.set_cell(x, y, current_player.choice)
+        if board.game_over
+          puts game_over_message
```
Comment (bellom) [inline lib/game.rb]: Game logic should be kept in classes (no “puts” and “get.chomp” inside the `game.rb`)

**Review by bellom — CHANGES_REQUESTED** [review]
Hi Team, 
Great work using `Git Flow`, just two things to do;
**1. fix the linter's errors.**
kindly fix all errors, checkout this [link](https://github.com/microverseinc/linters-config#troubleshooting) for help.
**2. Game logic should be kept in classes (no “puts” and “get.chomp” there!).**
kindly remove it from game logic.

_Kindly re-submit for review after making all those changes slightly_smiling_face Thumps -up await you._

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      board.set_cell(x, y, current_player.choice)
+      if board.game_over
+        puts game_over_message
+        board.formatted_grid
+        return
+      else
+        switch_players
+      end
+    end
+  end
```
Comment (maelfosso) [inline lib/game.rb]: No “puts” and “get.chomp” in class! Only in the *bin/main.rb* file. So, please, move this function out of that class.

File: lib/game.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    mapping[human_move]
+  end
+
+  public
+
+  def game_over_message
+    return "#{current_player.name} won!" if board.game_over == :winner
+    return "The game ended in a tie" if board.game_over == :draw
+  end
```
Comment (maelfosso) [inline lib/game.rb]: No “puts” and “get.chomp” in class! Only in the *bin/main.rb* file. I suggest you return `:winner` or `:draw` and you display the corresponding message in the *bin/main.rb* file

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello coders,

Nice implementation of OOP principles. Nice logic separation in class.
Please, correct error highlighted by Stickler and take a look of changes suggested.

Thanks

**Pull request closed without merging**